Created Data Layer and UI Layer. Cleaned UI warnings.

ldanielcano opened the pull request:
- Moved fragments and activities to the UI layer.
- Moved Movie.kt and Movies.kt to the datalayer
- Cleaned pending warnings from the UI.

ldanielcano committed:
Created datalayer and uilayer packages. Cleaned warnings.

estebanhiguitad reviewed (changes requested):
Agregar la capa de dominio. En la arquitectura es opcional, pero como objetivo de aprendizaje vamos a usarla.

ldanielcano committed:
Implemented the data Layer and a Domain MovieModel

ldanielcano reviewed (commented):
Confirm receipt

ldanielcano committed:
Created GetMoviesForUICase.kt and used it in the fragments

ldanielcano committed:
Created viewmodel and ui elements for UI Layer

Created livedata objects inside the MoviesViewModel.kt, for UI to be updated once there are changes in the MovieList.

Subscribed to those data objects in the UI elements SearchFragment.kt and HomeFragment.kt

Got rid of the resources requirement for MoviesLocalDataSourceImpl.kt

ldanielcano committed:
Changed the package location for some Activities

estebanhiguitad commented on IMDbApp/app/src/main/java/com/danielcano/imdbapp/uilayer/menu/SearchFragment.kt:
```diff
@@ -24,18 +25,22 @@ class SearchFragment : Fragment() {
+        val itemList = MoviesRepositoryImpl(MoviesLocalDataSourceImpl(this.resources)).getMovies()
         movieListAdapter.submitList(itemList)
-
+//        val moviesObserver = Observer<List<MovieModel>>{
```
eliminar código comentado

estebanhiguitad commented on IMDbApp/app/src/main/java/com/danielcano/imdbapp/uilayer/menu/HomeFragment.kt:
```diff
@@ -23,17 +27,23 @@ class HomeFragment: Fragment() {
+//        val itemList = movies(this.resources)
+        val itemList = MoviesRepositoryImpl(MoviesLocalDataSourceImpl(this.resources)).getMovies()
         movieListAdapter.submitList(itemList)
+//        val moviesObserver = Observer<List<MovieModel>>{
```
eliminar código comentado en todos los lugares que aplique en esta clase
